lib/rest-adapter.js

```diff
@@ -258,6 +259,10 @@ RestAdapter.prototype.createHandler = function() {
 
   var handleUnknownPaths = this._shouldHandleUnknownPaths();
 
+  // Mount rest-router
+  var customRouter = new RestRouter();
+  root.use(customRouter.handle());
```

Comment:
I see, `root.use` expects a handler function, while our router is an object instead.

You can use the approach used by express ([source](https://github.com/strongloop/express/blob/4fb3653b6ce9bd9080ceadb267f2cffc32413444/lib/router/index.js#L42-L60)), even though it's not very pretty.

```js
// lib/rest-router.js
var RestRouter = module.exports = function(options) {
  var opts = options || {};

  var router = function(req, res, next) {
    router.handle(req, res, next);
  }

  // mixin Router class functions
  router.__proto__ = RestRouter;

  router.trie = new Trie();
  router.options = options; 

  return router;
}
```